homeassistant/components/climacell/__init__.py

```diff
@@ -138,19 +162,61 @@ async def async_unload_entry(
     return unload_ok
 
 
+async def async_migrate_entry(
+    hass: HomeAssistantType, config_entry: ConfigEntry
+) -> bool:
+    """Migrate old entry."""
+    version = config_entry.version
+
+    _LOGGER.debug("Migrating from version %s", version)
+
+    # 1 -> 2: Added new config key to support multiple API versions and limited nowcast timesteps
+    if version == 1:
```

Comment:
Could we do this migration without changing entry version? Changing version makes it impossible to roll back. Maybe we csn just check for api version in entry data and migrate if the api version it's missing?

Reply:
Good idea